src/fparser/two/Fortran2008.py

```diff
@@ -96,6 +101,241 @@ class Program_Unit(Program_Unit_2003):  # R202
+        :rtype: None, Codimension_Attr_Spec
+        '''
+        return WORDClsBase.match(
+            'CODIMENSION', Coarray_Bracket_Spec, string, colons=False,
+            require_cls=True)
+
+
+class Coarray_Bracket_Spec(BracketBase):  # R502.d.0
+    """
+    <coarray-bracket-spec> = <lbracket> <coarray-spec> <rbracket>
+    """
+    subclass_names = []
+    use_names = ['Coarray_Spec']
+
+    @staticmethod
+    def match(string):
+        '''Implements the matching for the coarray specification
+        including the square brackets.
+
+        :param str string: the string to match as the specification.
+        :return: None if there is no match, otherwise the matched object.
+        :rtype: None, Coarray_Bracket_Spec
+        '''
+        return BracketBase.match('[]', Coarray_Spec, string)
+
+
+class Attr_Spec(STRINGBase):  # R502
+    """
+:F08R:`502`
+    <attr-spec> = <access-spec>
+                  | ALLOCATABLE
+                  | ASYNCHRONOUS
+                  | CODIMENSION <lbracket> <coarray-spec> <rbracket>
+                  | CONTIGUOUS
+                  | DIMENSION ( <array-spec> )
+                  | EXTERNAL
+                  | INTENT ( <intent-spec> )
+                  | INTRINSIC
+                  | <language-binding-spec>
+                  | OPTIONAL
+                  | PARAMETER
+                  | POINTER
+                  | PROTECTED
+                  | SAVE
+                  | TARGET
+                  | VALUE
+                  | VOLATILE
+    """
+    subclass_names = ['Access_Spec', 'Language_Binding_Spec',
+                      'Dimension_Attr_Spec', 'Intent_Attr_Spec',
+                      'Codimension_Attr_Spec']
+    use_names = []
+
+    @staticmethod
+    def match(string):
+        '''Implements the matching for a types attributes.
+
+        :param str string: the string to match as attribute.
+        :return None if there is no match, otherwise the matched object.
+        :rtype: None, Attr_Spec, Access_Spec, Language_Binding_Spec, \
+            Dimension_Attr_Spec, Intent_Attr_Spec, Codimension_Attr_Spec
+        '''
+        return STRINGBase.match(pattern.abs_attr_spec_f08, string)
+
+
+class Coarray_Spec(Base):  # R509
+    """
+:F08R:`509`
+    <coarray-spec> = <deferred-coshape-spec-list>
+                      | <explicit-coshape-spec-list>
+    """
+    subclass_names = ['Explicit_Coshape_Spec',
+                      'Deferred_Coshape_Spec_List']
+
+
+class Deferred_Coshape_Spec(SeparatorBase):  # R510
+    """
+:F08R:`510`
+    <deferred-coshape-spec> = :
+    """
+    subclass_names = []
+
+    @staticmethod
+    def match(string):
+        '''Implements the matching for deferred coarray shape specification.
+
+        :param str string: the string to match as deferred shape.
+        :return None if there is no match, otherwise the matched object.
+        :rtype: None, Deferred_Coshape_Spec
+        '''
+        if string == ':':
+            return None, None
```

Comment:
I think pylint like tuples to be specified within brackets these days i.e. `(None, None)`